### PR TITLE
Controllers testing

### PR DESCRIPTION
**Pull request by ericmahare**
In this section I did the following : -

- [ ] Add testing for user controller 
- [ ] Add testing for posts controller

### CODE REVIEW

**Review by rloterh — CHANGES_REQUESTED** [review]
Hi Eric,
 
 While you made a great effort in this project, unfortunately, I cannot proceed to review your code.
 
 ## Invalid Code Review Request
You have submitted a project, whereby Rubocop linter checks are failing on your Pull Request. Kindly ensure that **all** checks are passing.

**Rubocop lint error:**
image

 
Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏
 
 Cheers and Happy coding!👏👏👏
 
 Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
 
 **_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**
 
 _Invalid Code Review Request **does not count** into the code reviews limit._

**Comment by ericmahare** [comment]
Thanks for your review, I have fixed the linter error

File: spec/controllers/posts_spec.rb [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+RSpec.describe 'Posts', type: :request do
+  describe 'GET /index' do
+    before(:each) do
+      @user = User.new(name: 'Tom', postscounter: 0)
+      @post = Post.new(title: 'Title', text: 'text test', commentscounter: 1, likescounter: 0, user_id: 1)
+      @user.save
+      @post.save
+      get '/users/1/posts/2'
+    end
+    it 'renders correct template' do
```
Comment (Davidosky007) [inline spec/controllers/posts_spec.rb]: * Kindly add tests for your show actions as stated in the [requirements ](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/controller_specs_project.md#:~:text=Make%20sure%20that%20for,includes%20correct%20placeholder%20text.)
* Also i get these errors when i run the tests, Kindly fix them 
image

File: spec/controllers/user_spec.rb [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    it 'renders the index template' do
+      get '/users'
+      expect(response.status).to eq(200)
+      expect(response).to render_template('index')
+    end
+
+    it 'renders the index template' do
+      get '/users/1'
+      expect(response.status).to eq(200)
+      expect(response).to render_template('show')
```
Comment (Davidosky007) [inline spec/controllers/user_spec.rb]: * Kindly add tests for your show actions as stated in the [requirements](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/controller_specs_project.md#:~:text=Make%20sure%20that%20for,includes%20correct%20placeholder%20text.)

**Review by Davidosky007 — CHANGES_REQUESTED** [review]
Hi @ericmahare 👋, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
* Well written readme file 👍 
* No linter errors 👍 
* Proper use of Gitflow 👍 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @davidsoky007 in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/controllers/posts_spec.rb (hunk 1 above)
Comment (ericmahare) [reply spec/controllers/posts_spec.rb]: Hello @Davidosky007 all the tests are passing on my local machine what could be the issue can you please elaborate on that

Comment (Davidosky007) [reply spec/controllers/posts_spec.rb]: When i run `rspec` i get those errors from my end, i will suggest that you update your readme file with steps on how you were able to achieve this on your end

**Review by emmanuellekamwa — APPROVED** [review]
**STATUS: APPROVED 🍾✔🎉**
Hello @ericmahare 👋
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights

- You documented your work professionally.✔
- Your linters are passing all the checks
- You asked for the changes in a polite and friendly way.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**